### PR TITLE
Make new Registry page responsive

### PR DESCRIPTION
**Pull request by onishiweb**
This is a big PR mainly for a lot of structural changes to the way the Registry now gets it's styles and JS.

This PR:
- adds Gulp and OBT build process
- splits out the old CSS into multiple Sass partials
- breaks up the component detail view into multiple parts and adds them to a "blocks" folder
- adds header styles copied over from `o-techdocs`
- copies syntax highlighting from `o-techdocs`

Still to do is a full cleanup of all the styles and a lot of class renaming/restructuring/deleting. Also currently includes `o-overlay`, `o-buttons`, and `o-tabs` via `silent-mode: false` which needs to be changed to reduce the weight of the stylesheet.

Trello: https://trello.com/c/yjuXNnBB

### CODE REVIEW

**Comment by onishiweb** [comment]
Is merged into #28 would be good to review this briefly, but gets modified in the other branch.

File: app/views/blocks/component-data.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		<div class="val" data-o-grid-colspan="12 M8">
+			<span class="label-{% if is_valid %}pass{% endif %}{% if not is_valid %}fail{% endif %}">
+				{% if is_valid == 1 %}
+					OK
+				{% elseif is_valid == 0 %}
+					Not installable, fails build tests
+				{% else %}
+					Unknown
+				{% endif %}
+				<a style="margin-left: 5px; color: white" href="https://{{ SERVER.BUILD_SERVICE_HOST }}/modules/{{module_name}}@{{tag_name}}" title="View build test results"><i class="fa fa-info-circle"></i></a>
```
Comment (alicebartlett) [inline app/views/blocks/component-data.html:24]: Got some inline styles here presumably a large chunk of this code has been moved in from another file...

File: app/views/blocks/component-data.html [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		<div class="key" data-o-grid-colspan="12 M4">Last indexed</div>
+		<div class="val" data-o-grid-colspan="12 M8">
+			{{datetime_last_cached|date('j M Y H:i:s')}} (<a href="/components/{{module_name}}@{{tag_name}}/refresh" class="link-refresh">Build now</a>)
+		</div>
+	</div>
+	<div class="component-details-row">
+		<div class="key" data-o-grid-colspan="12 M4">Dependencies</div>
+		<div class="val" data-o-grid-colspan="12 M8">
+			{% if not dependencies %}
+				<span style="color:#bbb">None</span>
```
Comment (alicebartlett) [inline app/views/blocks/component-data.html:38]: another inline style

File: app/views/blocks/component-data.html [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				{% endfor %}
+			</table>
+			{% endif %}
+		</div>
+	</div>
+	<div class="component-details-row">
+		<div class="key" data-o-grid-colspan="12 M4">Dependents</div>
+		<div class="val" data-o-grid-colspan="12 M8">
+			{% if not dependents %}
+				<span style="color: #bbb">None</span>
```
Comment (alicebartlett) [inline app/views/blocks/component-data.html:64]: another inline style

File: public/css/registry.css [hunk 4]
```diff
@@ -384,3 +421,212 @@ body {
...
 }
+
+
+
+.component-navigation {
+	list-style: none;
+	padding-left: 0;
+}
+
+.component-navigation__group {
```
Comment (alicebartlett) [inline public/css/registry.css:432]: empty class here

**Pull request closed without merging**